### PR TITLE
fix(layout-grid): Prevent overflowing components within cells

### LINKED ISSUE #406: Image overflowing the cell div
### What MDC-Web Version are you using?

material-components-web@0.6.0

### What browser(s) is this bug affecting?

>This bug is affecting Firefox 52.
Mozilla/5.0 (Windows NT 10.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0

### What OS are you using?

>Windows 10 Version 1607

### What are the steps to reproduce the bug?

http://codepen.io/anon/pen/EWojwo

### What is the expected behavior?

>The image should not overflow the cell div 

### What is the actual behavior?

>The image overflow the cell div so it is superposing iwith the other cell div

### Any other information you believe would be useful?

>There is no problem on Chrome 56 and Edge 14. 
It seems to be due to a width: auto;.

### PR DESCRIPTION
**Pull request by vivekmarakana**
- Adding `max-height: 100%` and `max-width: 100%` to prevent overflowing components
- Also adding `overflow:hidden` in case it is still overflowing

Fixes #406

### CODE REVIEW

File: packages/mdc-layout-grid/_mixins.scss [hunk 1]
```diff
@@ -48,6 +48,9 @@
 
   @supports (display: grid) {
     width: auto;
+    max-width: 100%;
+    max-height: 100%;
+    overflow: hidden;
```
Comment (amsheehan) [inline packages/mdc-layout-grid/_mixins.scss]: @yeelan0319 Is this clipping the correct behavior? Do we want the image to be contained or covering the parent?

Comment (vivekmarakana) [inline packages/mdc-layout-grid/_mixins.scss]: I ran into an issue while fixing this. If there's any other component inside the cell, then the child cell will still overflow as it has 100% max-height.

Check out this pen: http://codepen.io/vivekmarakana/pen/VpxqJa

Comment (amsheehan) [inline packages/mdc-layout-grid/_mixins.scss]: That makes sense. What worries me about using `max-` width/height for this as a fix is that I can't see anyone being happy with their image being stretched. What if we use `min-` width/height? That way the aspect ratio of the image is preserved, even if the crop and resolution is unfavorable.

Would like to get @traviskaufman and @yeelan0319 to weigh in. 

Looks good so far though, thanks!

Comment (amsheehan) [inline packages/mdc-layout-grid/_mixins.scss]: We'd also want `overflow: hidden` on `mdc-layout-grid__cell`, no?

Comment (traviskaufman) [inline packages/mdc-layout-grid/_mixins.scss]: okay, so, backing _way_ up here:

What's the reason for attaching `mdc-layout-grid__cell` to the `` element within an additional `mdc-layout-grid__cell` element? Cells aren't supposed to be nested within other cells; that behavior is actually undefined.

**Comment by traviskaufman** [comment]
Closing as issue was fixed by a different solution.

**Pull request closed without merging**